### PR TITLE
Sampling Slider

### PR DESCRIPTION
**Pull request by Waidhoferj**
https://user-images.githubusercontent.com/33008436/129941379-17f27d35-5c45-4cfb-9b90-21afd849b38f.mov

## Changes
- Added `Slider` component based off of the `RangeSlider` component with some functional adjustments and simplifications.
- Added a Slider to the sampling dropdown in the `EditTab`
- Moved `Loader` to the `ui-widgets` folder
- Changed the `graph_data_config` object schema to more aptly named properties.

## Testing
1. Plot the Titanic dataset.
2. In the Graph Screen, adjust the slider and verify that the graph above updates accordingly

> Note that the slider only updates when it is released. This is to avoid infinite loading when users want to downsample a large dataset. Instantaneous updates for each intermittent slider value could cause the UI to lag.

### CODE REVIEW

File: src/components/Onboarding/OnboardingWidget.tsx [hunk 1]
```diff
@@ -36,9 +37,26 @@ export default function OnboardingWidget(props: OnboardingWidgetProps) {
...
+      graphConfig.datasetLength
+    );
+    setGraphConfig(
+      produce(graphConfig, (gc) => {
+        gc.sampleSize = 100;
+      })
+    );
     return () => {
-      setGraphConfig({ ...graphConfig, sample: false });
+      console.log('user defined?', userDefinedSample);
```
Comment (jahn96) [inline src/components/Onboarding/OnboardingWidget.tsx:52]: Remove this log

File: src/components/Sidebar/Tabs/EditTab.tsx [hunk 2]
```diff
@@ -435,6 +438,12 @@ export default function EditTab({
     }
   }
 
+  function setSamplingThreshold(val: number) {
+    console.log(val);
+    console.log(graphDataConfig);
```
Comment (jahn96) [inline src/components/Sidebar/Tabs/EditTab.tsx:443]: Remove log

File: src/widget.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ const bifrostModelPropDefaults = {
   plot_function_args: {} as Args,
   column_types: {} as Record<EncodingInfo['field'], EncodingInfo['type']>,
   column_name_map: {} as Record<string, string>,
-  graph_data_config: { maxRows: 100, sample: false } as GraphDataConfig,
+  graph_data_config: { sampleSize: 100, datasetLength: 1 } as GraphDataConfig,
```
Comment (jahn96) [inline src/widget.ts:47]: Is datasetLength the number of rows in the dataset?

**Review by jahn96 — CHANGES_REQUESTED** [review]
It looks like that the handle is misplaced initially. The graph describes the entire dataset, but handle is at 100 initially.
Screen Shot 2021-08-18 at 3 12 15 PM
Also, adding what percentage of the entire data has been sampled? 
Screen Shot 2021-08-18 at 3 15 18 PM

File: src/widget.ts (hunk 3 above)
Comment (Waidhoferj) [reply src/widget.ts:47]: Yes its the same as `len(df)`

**Review by jahn96 — APPROVED** [review]
LGTM!

**Pull request merged**